File: occasum-core/src/test/java/pw/stamina/occasum/properties/primitives/DoublePropertyTests.java
package pw.stamina.occasum.properties.primitives;

import pw.stamina.occasum.properties.PropertyTests;

import java.util.Random;

final class DoublePropertyTests extends PropertyTests<DoubleProperty> {

    @Override
    protected DoubleProperty provideProperty() {
        return DoubleProperty.from("test", 10);
    }

    @Override
    protected void shufflePropertyValue(DoubleProperty property, Random random) {
        property.set(random.nextDouble());
    }
}

File: occasum-core/src/test/kotlin/pw/stamina/occasum/properties/primitives/DoublePropertyTests.kt
package pw.stamina.occasum.properties.primitives

import pw.stamina.occasum.properties.PropertyTests
import java.util.*

internal class DoublePropertyTests : PropertyTests<DoubleProperty>() {

    override fun provideProperty(): DoubleProperty {
        return DoubleProperty("test", 10.0)
    }

    override fun shufflePropertyValue(property: DoubleProperty, random: Random) {
        property.value = random.nextDouble()
    }
}